scripts/coe_stats_gui.lua

```diff
@@ -0,0 +1,105 @@
+---@class coe.StatsGUI
+local StatsGUI = {}
+
+local mod_gui = require( "mod-gui" )
+
+-- =============================================================================
+
+local function build_stats_info_frame( statistics_frame )
+  local info_frame = statistics_frame.add{
+    type = "frame",
+    name = "coe_stats_info_frame",
+    direction = "vertical"
+  }
+
+  info_frame.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_title"}
+  }.parent.add{
+    type = "flow",
+    direction = "horizontal"
+  }.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_required"}
+  }.parent.add{
+    type = "label",
+    caption = tostring( global.rocket_silo.required_launches )
+  }.parent.parent.add{
+    type = "flow",
+    direction = "horizontal"
+  }.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_complete"}
+  }.parent.add{
+    type = "label",
+    caption = tostring( global.rocket_silo.total_launches )
+  }
+end -- build_stats_info_frame
+
+-------------------------------------------------------------------------------
+
+local function open_stats_ui(event)
+  local player = game.get_player(event.player_index)
+  local gui = player.gui.left
+  local statistics_frame =
+      gui.add {
+      type = "frame",
+      name = "coe_statistics_frame",
+      direction = "vertical",
+      caption = {"coe-stats-gui.label_stats_title"}
+  }
+
+  build_stats_info_frame( statistics_frame )
+
+  local pdata = global.players[event.player_index]
+  pdata.button_statistics_visible = true
```

Comment:
Whats the reason for storing the visible state of this window?

Reply:
using a tracking variable tied to the player is how the UI tutorial showed how to know if the UI was open.
I don't know if it is right.  I'll explore checking the UI elements directly.

Reply:
In some instance a tracking variable can be a good thing, but for a small on click gui that is created and destroyed I wouldnt bloat the global with it